tests/test_feature_selection/test_mrmr.py

```diff
@@ -0,0 +1,106 @@
+from unittest.mock import MagicMock
+
+import numpy as np
+import pytest
+from sklearn.datasets import make_classification, make_regression
+from sklearn.ensemble import RandomForestClassifier
+from sklearn.pipeline import Pipeline
+
+from sklego.common import flatten
+from sklego.feature_selection.mrmr import MaximumRelevanceMinimumRedundancy, _redundancy_pearson
+from tests.conftest import general_checks, transformer_checks
+
+
+@pytest.mark.parametrize("test_fn", flatten([general_checks, transformer_checks]))
+def test_transformer_checks(test_fn):
+    mrmr = MaximumRelevanceMinimumRedundancy(k=1)
+    test_fn(MaximumRelevanceMinimumRedundancy.__name__, mrmr)
+
+
+def test_redundancy_pearson():
+    """Test the pearson correlation for selected and left features:
+    For the 3rd column (index 2), it mirrors the values in the first and second columns (indexes 0, 1),
+    resulting in a correlation of 1 between the 3rd column and both the 1st and 2nd columns.
+    For the 4th column, the correlation with the 1st and 2nd columns is |(-1)|, which equals 1.
+    The total correlation sum is 4, accounting for all correlations.
+    """
+    X = np.array([[0.0, 0.0, 0.0, 1.0], [1.0, 1.0, 1.0, 0.0], [1.0, 1.0, 1.0, 0.0], [1.0, 1.0, 1.0, 0.0]])
+
+    selected = [0, 1]
+    left = [2, 3]
+    redundancy_scores = _redundancy_pearson(X, selected, left)
+    redundancy_scores_expected = np.array([2 * 1.0, 2 * 1.0])
+    assert np.allclose(redundancy_scores, redundancy_scores_expected, atol=0.00001)
+
+
+@pytest.mark.parametrize("k", [1.2, 5, -1])
+def test_mrmr_raises(k):
+    X, y = make_classification(n_features=4)
+    with pytest.raises(ValueError):
+        _ = MaximumRelevanceMinimumRedundancy(k=k).fit(X, y)
```

Comment:
Nitpick, but why have the `_` here? 

```suggestion
        MaximumRelevanceMinimumRedundancy(k=k).fit(X, y)
```